shared/core/src/fixed_vec.rs

```diff
@@ -7,6 +7,7 @@ use ts_rs::TS;
 
 #[derive(Clone, Copy, Zeroable, AnchorSerialize, AnchorDeserialize, TS)]
 #[ts(type = "Array<T>", bound = "T: TS")]
+#[repr(C)]
```

Comment:
Same as the above, this used to be a `Repr(Rust)` but by sheer luck this is binary equivalent to `Repr(C)` in the current rust implementation, switching to `Repr(C)` to ensure the layout doesn't change when we switch rust compiler version